### PR TITLE
update sql scripts with new ascn table

### PR DESCRIPTION
**Pull request by averyniceday**
ASCN sql changes. Haven't tested it yet - going to go through your code now! 
# Checks
- [ ] Runs on Heroku.
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Google Style Guide](https://github.com/google/styleguide).
- [ ] If this is a feature, the PR is to rc. If this is a bug fix, the PR is to master.
- [ ] If this is a bug fix, create a unit and/or e2e test, or explain why that cannnot be done.

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).

### CODE REVIEW

File: core/src/main/java/org/mskcc/cbio/portal/dao/DaoAlleleSpecificCopyNumber.java [hunk 1]
```diff
@@ -0,0 +1,1607 @@
...
+    public static final String NAN = "NaN";
+    private static final String MUTATION_COUNT_ATTR_ID = "MUTATION_COUNT";
+
+    public static long getLargestAscnId() throws DaoException {
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+        try {
+            con = JdbcUtil.getDbConnection(DaoAlleleSpecificCopyNumber.class);
+            pstmt = con.prepareStatement("SELECT MAX(`ASCB_ID`) FROM `allele_specific_copy_number`");
```
Comment (ao508) [inline core/src/main/java/org/mskcc/cbio/portal/dao/DaoAlleleSpecificCopyNumber.java]: >ASCB_ID

typo?

File: core/src/main/java/org/mskcc/cbio/portal/dao/DaoAlleleSpecificCopyNumber.java [hunk 2]
```diff
@@ -0,0 +1,1607 @@
...
+            rs = pstmt.executeQuery();
+            return rs.next() ? rs.getLong(1) : 0;
+        } catch (SQLException e) {
+            throw new DaoException(e);
+        } finally {
+            JdbcUtil.closeAll(DaoAlleleSpecificCopyNumber.class, con, pstmt, rs);
+        }
+    }
+    // ***================ EVERYTHING UNDER HERE IS SUBJECT TO CHANGE ====================***
```
Comment (ao508) [inline core/src/main/java/org/mskcc/cbio/portal/dao/DaoAlleleSpecificCopyNumber.java]: why are DaoMutation methods in here?

File: core/src/main/java/org/mskcc/cbio/portal/model/AlleleSpecificCopyNumber.java [hunk 3]
```diff
@@ -0,0 +1,689 @@
+/*
+ * Copyright (c) 2019 Memorial Sloan-Kettering Cancer Center.
```
Comment (ao508) [inline core/src/main/java/org/mskcc/cbio/portal/model/AlleleSpecificCopyNumber.java:2]: Mutation objects should remain in their own class

File: db-scripts/src/main/resources/cgds.sql [hunk 4]
```diff
@@ -840,7 +842,20 @@ CREATE TABLE `data_access_tokens` (
...
+    `CLONAL` boolean DEFAULT NULL,
+    `MINOR_COPY_NUMBER` int DEFAULT NULL,
+    `MUTANT_COPIES` int DEFAULT NULL,
+    `TOTAL_COPY_NUMBER` int DEFAULT NULL,
+    PRIMARY KEY (`ASCN_ID`),
+    FOREIGN KEY (`ASCN_ID`) REFERENCES `mutation` (`ASCN_ID`) ON DELETE CASCADE
+);
 -- --------------------------------------------------------
 CREATE TABLE `info` (
   `DB_SCHEMA_VERSION` varchar(24),
```
Comment (ao508) [inline db-scripts/src/main/resources/cgds.sql:862]: need to update the version number at bottom of this script bc we're making a db change

File: db-scripts/src/main/resources/migration.sql [hunk 5]
```diff
@@ -715,3 +715,24 @@ UPDATE `cancer_study`
...
+    `CCF_M_COPIES` float DEFAULT NULL,
+    `CLONAL` boolean DEFAULT NULL,
+    `MINOR_COPY_NUMBER` int DEFAULT NULL,
+    `MUTANT_COPIES` int DEFAULT NULL,
+    `TOTAL_COPY_NUMBER` int DEFAULT NULL,
+    PRIMARY KEY (`ASCN_ID`),
+    FOREIGN KEY (`ASCN_ID`) REFERENCES `mutation` (`ASCN_ID`) ON DELETE CASCADE
+);
+
+UPDATE `info` SET `DB_SCHEMA_VERSION`="3.1.0";
```
Comment (ao508) [inline db-scripts/src/main/resources/migration.sql]: cgds.sql needs the updated version number as does the master pom.xml

**Review by ao508 — CHANGES_REQUESTED** [review]

File: db-scripts/src/main/resources/migration.sql [hunk 6]
```diff
@@ -715,3 +715,24 @@ UPDATE `cancer_study`
     INNER JOIN `mutation_event` ON `mutation`.MUTATION_EVENT_ID = `mutation_event`.MUTATION_EVENT_ID
 SET `cancer_study`.REFERENCE_GENOME_ID = IF(`mutation_event`.NCBI_BUILD in ('37', 'hg19','GRCh37'), 1, 2);
 UPDATE `info` SET `DB_SCHEMA_VERSION`="2.11.0";
+
+##version: 3.1.0
+ALTER TABLE `mutation_event` ADD COLUMN ASCN_ID int DEFAULT NULL;
+ALTER TABLE `mutation_event` ADD COLUMN ANNOTATION_JSON JSON DEFAULT NULL;
+-- ========================== new ascn tables =============================================
```
Comment (ao508) [inline db-scripts/src/main/resources/migration.sql]: >tables
--> `table`

**Comment by averyniceday** [comment]
oops I've totally been pushing dev work into the same branch!! I meant to only open a PR with the sql changes! 😅

File: core/src/main/java/org/mskcc/cbio/portal/model/AlleleSpecificCopyNumber.java [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+package org.mskcc.cbio.portal.model;
+
+import java.util.*;
+
+/**
+ * Encapsulates Details regarding a Single Mutation.
+ *
+ * @author averyniceday.
+ */
+public final class AlleleSpecificCopyNumber {
```
Comment (ao508) [inline core/src/main/java/org/mskcc/cbio/portal/model/AlleleSpecificCopyNumber.java]: add `implements Serializable``` here

Comment (ao508) [reply core/src/main/java/org/mskcc/cbio/portal/model/AlleleSpecificCopyNumber.java]: also, remove the `final` here as well

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java [hunk 8]
```diff
@@ -467,13 +467,13 @@ public void importData() throws IOException, DaoException {
...
         }
-        // immport ascn records first since mutations reference ascn db table
-        //for (AlleleSpecificCopyNumber ascn : ascnRecords) {
-        //  try {
-        //        DaoAlleleSpecificCopyNumber.addAscn(ascn);
-        //  } catch (DaoException ex) {
-        //        throw ex;
-        //  }
+ 
+        for (AlleleSpecificCopyNumber ascn : ascnRecords) {
```
Comment (ao508) [inline core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java:471]: looks like some whitespace clean up here :)

**Review by ao508 — CHANGES_REQUESTED** [review]
a few more changes to make and then this is ready to test IMO!

File: core/src/main/java/org/mskcc/cbio/portal/dao/DaoAlleleSpecificCopyNumber.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2015 Memorial Sloan-Kettering Cancer Center.
```
Comment (sheridancbio) [inline core/src/main/java/org/mskcc/cbio/portal/dao/DaoAlleleSpecificCopyNumber.java]: year needs update

File: core/src/main/java/org/mskcc/cbio/portal/model/AlleleSpecificCopyNumber.java [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package org.mskcc.cbio.portal.model;
+
+import java.util.*;
+
+/**
+ * Encapsulates Details regarding a Single Mutation.
```
Comment (sheridancbio) [inline core/src/main/java/org/mskcc/cbio/portal/model/AlleleSpecificCopyNumber.java]: maybe update class description comment

File: core/src/main/java/org/mskcc/cbio/portal/model/AlleleSpecificCopyNumber.java [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+ *
+ * @author averyniceday.
+ */
+public final class AlleleSpecificCopyNumber {
+
+    private long ascnId;
```
Comment (sheridancbio) [inline core/src/main/java/org/mskcc/cbio/portal/model/AlleleSpecificCopyNumber.java:43]: The other numeric type members do not use primitive classes .. they use Integer and Float ... should we use Long to be consistent?

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java [hunk 12]
```diff
@@ -114,10 +114,10 @@ public void importData() throws IOException, DaoException {
         Set<MutationEvent> newEvents = new HashSet<MutationEvent>();
 
         Map<ExtendedMutation,ExtendedMutation> mutations = new HashMap<ExtendedMutation,ExtendedMutation>();
-        // long largestAscnId = DaoAlleleSpecificCopyNumber.getLargestAscnId();
-        // List<AlleleSpecificCopyNumber> ascnRecords = new ArrayList<AlleleSpecificCopyNumber>();
-
         long mutationEventId = DaoMutation.getLargestMutationEventId();
+
```
Comment (sheridancbio) [inline core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java:118]: code updates here, so copyright can become "Copyright (c) 2015 - 2019"

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java [hunk 13]
```diff
@@ -467,13 +467,13 @@ public void importData() throws IOException, DaoException {
...
-        //  try {
-        //        DaoAlleleSpecificCopyNumber.addAscn(ascn);
-        //  } catch (DaoException ex) {
-        //        throw ex;
-        //  }
+ 
+        for (AlleleSpecificCopyNumber ascn : ascnRecords) { 
+            try {
+                DaoAlleleSpecificCopyNumber.addAlleleSpecificCopyNumber(ascn);
+            } catch (DaoException ex) {
```
Comment (sheridancbio) [inline core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java:474]: I'm not sure what the reason is for catching the exception here and then rethrowing it

File: db-scripts/src/main/resources/cgds.sql [hunk 14]
```diff
@@ -498,6 +498,8 @@ CREATE TABLE `mutation` (
   `DRIVER_FILTER_ANNOTATION` VARCHAR(80),
   `DRIVER_TIERS_FILTER` VARCHAR(50),
   `DRIVER_TIERS_FILTER_ANNOTATION` VARCHAR(80),
+  `ASCN_ID` int(11),
```
Comment (sheridancbio) [inline db-scripts/src/main/resources/cgds.sql]: I think we decided on BigInt for this

File: db-scripts/src/main/resources/cgds.sql [hunk 15]
```diff
@@ -840,11 +842,24 @@ CREATE TABLE `data_access_tokens` (
     PRIMARY KEY (`TOKEN`),
     FOREIGN KEY (`USERNAME`) REFERENCES `users` (`EMAIL`) ON DELETE CASCADE
 );
-
+-- --------------------------------------------------------
+CREATE TABLE `allele_specific_copy_number` (
+    `ASCN_ID` int(255) NOT NULL auto_increment,
```
Comment (sheridancbio) [inline db-scripts/src/main/resources/cgds.sql]: BigInt?

File: db-scripts/src/main/resources/cgds.sql [hunk 16]
```diff
@@ -840,11 +842,24 @@ CREATE TABLE `data_access_tokens` (
...
+    FOREIGN KEY (`ASCN_ID`) REFERENCES `mutation` (`ASCN_ID`) ON DELETE CASCADE
+);
 -- --------------------------------------------------------
 CREATE TABLE `info` (
   `DB_SCHEMA_VERSION` varchar(24),
   `GENESET_VERSION` varchar(24)
 );
 -- THIS MUST BE KEPT IN SYNC WITH db.version PROPERTY IN pom.xml
-INSERT INTO info VALUES ('2.11.0', NULL);
+INSERT INTO info VALUES ('3.1.0', NULL);
```
Comment (sheridancbio) [inline db-scripts/src/main/resources/cgds.sql]: we talked about major v. minor schema version change

**Review by sheridancbio — APPROVED** [review]
overall looks good

**Pull request merged**